pkg/dockerregistry/server/auth.go

```diff
@@ -229,45 +253,61 @@ func (ac *tokenAuthChallenge) SetHeaders(w http.ResponseWriter) {
 	w.Header().Set("WWW-Authenticate", str)
 }
 
+func (ac *AccessController) logAndWrapErr(ctx context.Context, err error) error {
```

Comment:
logAndWrapErr is a weird coupling, especially when it only actually logs one type of error